spinalcordtoolbox/reports/qc.py

```diff
@@ -210,8 +210,8 @@ def label_vertebrae(self, mask, ax):
                     x += 10
                     label = list(self._labels_regions.keys())[list(self._labels_regions.values()).index(index)]
                     ax.text(x, y, label, color='black', clip_on=True)
-                    x -= 0.5
-                    y -= 0.5
+                    x -= y/500
+                    y -= y/500
```

Comment:
`patheffects` solution looks good to me! 👍🏻 I agree that it will handle the shadows more reliably across different image sizes than the current manual shift.

Reply:
Thank you! Tested for `sct_qc.py -p sct_label_vertebrae ...` and works fine for sagittal as well as axial input data.

During the testing, I have tried also another variant of `Sagittal` QC -- `sct_qc.py -p sct_label_utils ...`. And I have noticed that the numbers are hardly readable. Thus, I tried to apply your `patheffects` workaround.

Orginal (without `patheffects`):
<img width="430" alt="image" src="https://user-images.githubusercontent.com/39456460/192596278-c7168df1-2465-44e8-af5f-a235782566b1.png">

With `patheffects`:
<img width="431" alt="image" src="https://user-images.githubusercontent.com/39456460/192596378-b3d46f89-8338-449f-b44f-f13941c1f5df.png">

I would say that also here are number more readable with the `patheffects`. Thus, I have added `patheffects` also for `label_utils` in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3889/commits/1e221526ef3398ee7d5a1071a3130c6c71cca000

One more thing I have noticed; the horizontal offset for the `label_utils` is computed from the input image:
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/483355f2c1fbd62c3b44ab28203239487011c134/spinalcordtoolbox/reports/qc.py#L180

But the horizontal offset for the `label_vertebrae` is hardcoded:
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/483355f2c1fbd62c3b44ab28203239487011c134/spinalcordtoolbox/reports/qc.py#L211

We could probably unify that. I tried horizontal offset computed from the input image also for `label_vertebrae` and it looks fine. Or, what do you thing?

Reply:
> We could probably unify that. I tried horizontal offset computed from the input image also for label_vertebrae and it looks fine. Or, what do you thing?

I think this is a good idea! But, maybe with a different divisor? 

`make.shape[1] / 50` works for `label_utils` because they're point labels, meaning they're not as wide, and need less of an offset. But, with vertebral labels, they're thicker regions, so they might need more of an offset?

For example, `sct_example_data/t2/t2.nii.gz` has a width of 320, so using 50 will result in the label being a bit closer to the segmentation:

![ezgif-2-2b31821115](https://user-images.githubusercontent.com/16181459/192612460-a323db1d-915d-492e-b93a-b38654a8a593.gif)

Experimentally, I think that dividing by 25 produces nice results?

![image](https://user-images.githubusercontent.com/16181459/192614299-755a9dcb-51de-4771-9840-9e1d65fff4e0.png)

![image](https://user-images.githubusercontent.com/16181459/192614323-c752274f-79c1-4ee6-9c29-ec2b2d519c06.png)

Reply:
One last thing: I think the colormap for the vertebral labels isn't the best? The dark purple looks almost unreadable to me. So, I think I might remove that

Reply:
I've swapped out the colors so that:

* There's a high enough contrast with black (>5:1, via https://webaim.org/resources/contrastchecker/)
* The colors are distinguishable for people will colorblindness (via https://www.color-blindness.com/coblis-color-blindness-simulator/)

It now looks like:

![Screenshot from 2022-09-27 16-34-20](https://user-images.githubusercontent.com/16181459/192630681-d907be55-6020-4e21-893f-2f85146b0b9e.png)

Reply:
Looks great!

Reply:
> 
> `linewidth=1` looks very nice, though:
> 
> ![image](https://user-images.githubusercontent.com/16181459/192608552-b1dea98a-394b-40ce-9aae-de05e43db72c.png)
> 
> I've added this in [cd730dc](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3889/commits/cd730dc1f26006b3ffded02b6e15915a0151a37f).

Agree, `linewidth=1` looks nice 👍🏻

Reply:
> 
> Experimentally, I think that dividing by 25 produces nice results?
> 

Agree. `x += data.shape[1] / 25` works fine for me for both axial as well as sagittal input image.

Reply:
> > Experimentally, I think that dividing by 25 produces nice results?
> 
> Agree. `x += data.shape[1] / 25` works fine for me for both axial as well as sagittal input image.

Whoops! I accidentally committed this change as part of [480159e](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3889/commits/480159e74c8888fd395210ba90563b2a8de9b4ad), instead of its own commit.

All of these commits will be squashed anyway, though, so I might not fix it. :sweat_smile: